src/slave/containerizer/mesos/isolators/namespaces/ipc.cpp

```diff
@@ -295,8 +302,8 @@ Future<Option<ContainerLaunchInfo>> NamespacesIPCIsolatorProcess::prepare(
               "tmpfs",
               MS_NOSUID | MS_NODEV | MS_STRICTATIME,
               shmSize.isSome() ?
-                strings::format("mode=1777,size=%d", shmSize->bytes()).get() :
-                "mode=1777");
+                strings::format("mode=0700,size=%d,uid=%s", shmSize->bytes(), user).get() :
+                strings::format("mode=0700,uid=%s", user).get());
```

Comment:
OPT
Better: format "mode=xxx,uid=xxx" once and use it everywhere directly.
For that last occurence it can be used in that format directly (order of parameters doesn't count)